### DISCUSSION TITLE
use filenames as labels

### DISCUSSION THREAD

**Discussion by zatome** [ImageMagick/ImageMagick#2344] [General]
Hi all,
I try to convert a group of jpg or png files into new files using their filenames to create labels on top of the images.
I found this code but I can't get what I want.

`convert file.jpg -gravity Center -background orange label:[filename] +swap -append file.jpg`

I've tried with just one file and it works, except for the fact that I get an error message (freetype library missing).
Any idea to apply this to multiple files at once ?
Thanks

**Comment by snibgo** [comment]
Your freetype problem needs to be fixed. What is the complete text output from "convert -version"?

**Reply by zatome** [reply]
Thanks you for your help.
I get the following message 

> Version: ImageMagick 6.9.1-0 Q16 x86_64 2015-03-22 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2015 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Features: DPC OpenCL
Delegates (built-in): bzlib fftw jng jpeg lcms lzma png tiff webp xml zlib

I'm on os x 10.14
I realize my IM version is an old one :-)

**Reply by fmw42** [reply]
You need to install the freetype delegate. I would suggest you just upgrade your ImageMagick version as well.

**Reply by zatome** [reply]
I installed freetype, tried 
`brew link --overwrite freetype` 
and even  
`brew unlink freetype && brew link freetype` 
and I get this message, 
`Warning: Already linked: /usr/local/Cellar/freetype/2.10.2`
So link seems to be made.

When upgrading ImageMagick with brew, I get this
```
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
Updated 5 formulae.

Error: imagemagick not installed
```

Does that mean I have more than one IM version installed ? 
Another version that makes a mess ?

**Reply by fmw42** [reply]
Sorry, I do not know homebrew.  What do you get from the command

type -a convert

and 

type -a magick

**Reply by zatome** [reply]
type -a convert
`convert is /opt/ImageMagick/bin/convert`

type -a magick
`-bash: type: magick: not found`

**Reply by fmw42** [reply]
Looks like you only have one version and it is ImageMagick 6. Again I do not know homebrew. But I would suggest de-installing and then install again. Be sure to install with freetype if it does not list in your delegates. See if you can find out what delegates are installed with Homebrew ImageMagick. Search Google for help with de-installing and installing ImageMagick with Homebrew.

**Comment by snibgo** [comment]
It that a Mac computer? Sorry, I don't know how to upgrade IM on a Mac. It probably needs you to install freetype, then re-build IM. I suggest you search this forum for posts by user "fmw42".

**Reply by zatome** [reply]
Yes, it's a Mac :-)
Ok I'll do that, but do you have any idea anyway about applying "convert" to multiple files as I tried to do ?

**Comment by snibgo** [comment]
> ... applying "convert" to multiple files ...

I would do that with a shell loop. For example, in bash:
```
mkdir outdir

arr=(`ls x*.png`)

numimg=${#arr[*]}

for ((i=0; i<numimg; i++)); do
  echo $i ${arr[$i]}

  convert ${arr[$i]} -gravity Center -background orange label:${arr[$i]} +swap -append outdir/${arr[$i]}
done
```

**Comment by zatome** [comment]
Thanks, it works fine ! I just removed the "x" before .png.
What if I want to keep filename without extension on the label ?

**Comment by snibgo** [comment]
Sorry, I don't know. Something in bash could do that. (I know the answer for Windows BAT: %~dpn1 or whatever.)

**Reply by fmw42** [reply]
This works for me on my Mac in IM 6.9.11.24 Q16

`convert lena.png -set option:fn "%t" -gravity Center -background orange label:"%[fn]" +swap -append x.png
`

![x](https://user-images.githubusercontent.com/13458218/89060974-c837d700-d318-11ea-8ebb-292afc17d220.png)

**Reply by zatome** [reply]
![x](https://user-images.githubusercontent.com/50579373/89061545-2a1e2e00-d365-11ea-8e58-28db33fa0f31.png)

It works for me too, but I still have some error messages

```
convert: delegate library support not built-in `none' (Freetype) @ warning/annotate.c/RenderFreetype/1560.
convert: delegate library support not built-in `none' (Freetype) @ warning/annotate.c/RenderFreetype/1560.
convert: delegate library support not built-in `none' (Freetype) @ warning/annotate.c/RenderFreetype/1560.
convert: delegate library support not built-in `none' (Freetype) @ warning/annotate.c/RenderFreetype/1560.
convert: clBuildProgram failed. (-11) @ warning/opencl.c/CompileOpenCLKernels/898.
convert: clCreateCommandQueue failed. (0) @ warning/opencl.c/InitOpenCLEnvInternal/1211.
```

**Reply by fmw42** [reply]
To avoid those warnings, you need to have Freetype installed such that ImageMagick can find it and list it in convert -version.
